staging/src/k8s.io/component-base/metrics/registry.go

```diff
@@ -49,6 +49,29 @@ func shouldHide(currentVersion *semver.Version, deprecatedVersion *semver.Versio
 	return false
 }
 
+func validateShowHiddenMetricsVersion(currentVersion semver.Version, targetVersionStr string) error {
+	if targetVersionStr == "" {
+		return nil
+	}
+
+	validVersionStr := fmt.Sprintf("%d.%d", currentVersion.Major, currentVersion.Minor-1)
```

Comment:
I don't think we're ever going to have a v2, but if we did, it would print 2.-1 which is not right :)

Feel free to fix in a follow-up if you want.